apps/gp2-server/test/data-providers/user.data-provider.test.ts

```diff
@@ -290,6 +301,114 @@ describe('User data provider', () => {
     );
   });
 
+  describe('Create', () => {
+    afterEach(() => {
+      expect(nock.isDone()).toBe(true);
+    });
+
+    afterEach(() => {
+      nock.cleanAll();
+    });
+
+    test('Should throw when the POST request to squidex fails', async () => {
+      nock(baseUrl)
+        .post(`/api/content/${appName}/users?publish=true`)
+        .reply(500);
+
+      await expect(
+        userDataProvider.create(getUserCreateDataObject()),
+      ).rejects.toThrow(GenericError);
+    });
+
+    test('Should create the user', async () => {
+      const userResponse = fetchUserResponse();
+      const userCreateDataObject = getUserCreateDataObject();
+
+      nock(baseUrl)
+        .post(`/api/content/${appName}/users?publish=true`, getUserInput())
+        .reply(200, userResponse);
+
+      const response = await userDataProvider.create(userCreateDataObject);
+
+      expect(response).toEqual(userResponse.id);
+    });
+
+    test.each`
+      region                      | expected
+      ${'Africa'}                 | ${UsersDataRegionEnum.Africa}
+      ${'Asia'}                   | ${UsersDataRegionEnum.Asia}
+      ${'Australia/Australiasia'} | ${UsersDataRegionEnum.AustraliaAustraliasia}
+      ${'Europe'}                 | ${UsersDataRegionEnum.Europe}
+      ${'North America'}          | ${UsersDataRegionEnum.NorthAmerica}
+      ${'South America'}          | ${UsersDataRegionEnum.SouthAmerica}
+      ${'Latin America'}          | ${UsersDataRegionEnum.LatinAmerica}
```

Comment:
We shouldn't use the GraphQL Enum. We've typed them again in the squidex package.